core/data/core.joke

```diff
@@ -4189,3 +4196,190 @@
   {:added "1.0"}
   []
   (joker-version__))
+
+(defn- check-valid-options
+  "Throws an exception if the given option map contains keys not listed
+  as valid, else returns nil."
+  [options & valid-keys]
+  (when (seq (apply disj (apply hash-set (keys options)) valid-keys))
+    (throw
+      (ex-info ; was: (IllegalArgumentException.
+        (apply str "Only these options are valid: "
+          (first valid-keys)
+          (map #(str ", " %) (rest valid-keys))) {}))))
+
+;;multimethods
+(defn- space-str
+  "Quick and dirty version of joker.string/join with a single space as the separator, because joker.core probably shouldn't depend on joker.string"
+  ([] "")
+  ([a] (str " " a))
+  ([a & more] (str " " a (apply space-str more))))
+
+(defn- multifn-nomatch
+  [name args]
+  (throw (ex-info (str "No method in multimethod for dispatch value: (" name (apply space-str args) ")") {})))
+
+(defn- new-multifn
```

Comment:
I would prefer the name `multimethod__` for this function.

Reply:
Done.